### ISSUE TITLE
cann't modify the data members of a c++ object passed by value

### ISSUE THREAD

**Issue by sophiepeithos** [vinniefalco/LuaBridge#177]

**Comment by sophiepeithos** [comment]
```
template <class T>
struct DataClass
{
  DataClass (T data)
    : data (data)
  {
  }

  static void Test(DataClass& object)
  {
	  object.data = 20;
  }

  mutable T data;
};

using IntClass = DataClass <int>;
luabridge::getGlobalNamespace (L)
  .beginClass <IntClass> ("IntClass")
  .addConstructor <void (*) (int)> ()
  .addStaticFunction("test", &IntClass::Test)
  .endClass ();
runLua("result = IntClass (501)");
runLua("IntClass.test(result)");
ASSERT_TRUE(result()["data"].isNumber());
ASSERT_EQ(20, result()["data"].cast <int>());
```

**Comment by dmitry-t** [comment]
This should have been fixed by #189.
